tokio-util/src/codec/mod.rs

```diff
@@ -7,6 +7,69 @@
 //! [`AsyncWrite`], to framed streams implementing [`Sink`] and [`Stream`].
 //! Framed streams are also known as transports.
 //!
+//! # Example encoding using `LinesCodec`
+//! The following example demonstrates how to use a codec such as [`LinesCodec`] to
+//! write a sink of framed data. [`FramedWrite`] can be used to achieve this. Data sent
+//! to [`FramedWrite`] are first framed according to a specific codec, and then sent to
+//! an implementor of [`AsyncWrite`].
+//! ```
```

Comment:
We generally have empty lines between headers, text, and code blocks.
```suggestion
//! # Example encoding using `LinesCodec`
//!
//! The following example demonstrates how to use a codec such as [`LinesCodec`] to
//! write a sink of framed data. [`FramedWrite`] can be used to achieve this. Data sent
//! to [`FramedWrite`] are first framed according to a specific codec, and then sent to
//! an implementor of [`AsyncWrite`].
//!
//! ```
```